### PR TITLE
makes keylistener hook-based and mocks `Byond` a bit

### PR DESCRIPTION
**Pull request by ZeWaka**
# depends on #36

## About the PR 

title

## Why's this needed? 

more modern component definition

## Is there a relevant [tgui-styles](https://github.com/tgstation/tgui-styles) PR associated with this one?

no

### CODE REVIEW

File: lib/components/KeyListener.tsx [hunk 1]
```diff
@@ -8,32 +8,26 @@ type KeyListenerProps = Partial<{
...
-export class KeyListener extends Component<KeyListenerProps> {
-  dispose: () => void;
-
-  constructor(props) {
-    super(props);
-
-    this.dispose = listenForKeyEvents((key) => {
-      if (this.props.onKey) {
-        this.props.onKey(key);
+export const KeyListener = (props: KeyListenerProps) => {
```
Comment (jlsnow301) [inline lib/components/KeyListener.tsx:11]: booo make it a function

File: stories/KeyListener.stories.tsx [hunk 2]
```diff
@@ -11,8 +13,18 @@ export default {
...
+export const onKeyDown: Story = {
+  render: () => {
+    useEffect(() => {
+      setupHotKeys();
+    }, []);
+
+    return (
+      <KeyListener
+        onKeyDown={(e: KeyEvent) => {
+          console.log('onKeyDown', e.toString(), e);
```
Comment (jlsnow301) [inline stories/KeyListener.stories.tsx:25]: Make sure to add the withConsole decorator like input has

Comment (ZeWaka) [reply stories/KeyListener.stories.tsx:25]: Not necessary

Comment (jlsnow301) [reply stories/KeyListener.stories.tsx:25]: Where are the console logs showing?

**Pull request merged**